cleanlab_studio/internal/dataset_source/pandas_dataset_source.py

```diff
@@ -0,0 +1,26 @@
+import io
+import os
+from typing import IO, Any
+import pandas as pd
+
+from .dataset_source import DatasetSource
+
+
+class PandasDatasetSource(DatasetSource):
+    def __init__(self, *args: Any, df: pd.DataFrame, name: str, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.name = name
+        self._fileobj = self._init_fileobj_from_df(df)
+        self.file_size = self._get_size_in_bytes()
+        self.file_type = "text/csv"
+
+    def _init_fileobj_from_df(self, df: pd.DataFrame) -> IO[bytes]:
+        fileobj = io.BytesIO()
+        df.to_csv(fileobj)
+        fileobj.seek(0)
+        return fileobj
+
+    def _get_size_in_bytes(self) -> int:
+        with self.fileobj() as dataset_file:
+            dataset_file.seek(0, os.SEEK_END)
+            return dataset_file.tell()
```

Comment:
will this not leave the cursor at the end of the file? is this an issue?

Reply:
Oops yeah I meant to seek back to 0 before returning. Good catch!

Reply:
Actually this is also fine as is because the `self.fileobj()` context manager contains a `finally: fileobj.seek(0)`

Reply:
got it, didn't see that! thx